src/prefect/server/database/configurations.py

```diff
@@ -118,6 +121,9 @@ def __init__(
         connection_timeout: Optional[float] = None,
         sqlalchemy_pool_size: Optional[int] = None,
         sqlalchemy_max_overflow: Optional[int] = None,
+        statement_cache_size: Optional[int] = None,
+        prepared_statement_cache_size: Optional[int] = None,
+        application_name: Optional[str] = None,
```

Comment:
Since these are only used with Postgres, can you add these attributes to `AsyncPostgresConfiguration` instead of the base class?

Reply:
Thanks for the review @desertaxle! 

Sure, commit 695bd0 moves these settings to AsyncPostgresConfiguration.


Regarding implementation, I see two possible approaches:

1. Current approach: Extend `__init__` with additional parameters.
2. Bypassing base settings from `AsyncPostgresConfiguration` to `BaseDatabaseConfiguration`using kwargs, but adding `DatabaseConfigKwargs as TypedDict` to ensure sync between them.


I've implemented approach number 1 since I don't know which patterns for this codebase I should follow :) 

But I'm open to switching to the TypedDict approach if you think it would be more maintainable in the long term.

Reply:
Hi @desertaxle, commit [1695bd0](https://github.com/PrefectHQ/prefect/pull/16374/commits/1695bd098379c57ca1e366ad74de0b5c49576267) has applied your suggestions. 

When you have time, could you take a look at this comment? https://github.com/PrefectHQ/prefect/pull/16374#issuecomment-2545404034

Reply:
@williamjamir can you try reinstalling your dependencies and re-running the hooks? It looks like the discrepancy might be from and older version of `pydantic`.

Reply:
Thank you, @desertaxle! That really did the trick. The commit [c540f8b](https://github.com/PrefectHQ/prefect/pull/16374/commits/c540f8b16c19921e10797add2c0d271ba5615094) fixed the lint issue.